Bump to 4.2.1 and Add the bump version helper (#219)
nimblehq/elixir-templates: README.md

The release process follows the [Git flow](https://nimblehq.co/compass/development/version-control/release-management).

> 1. Bump the new version on local

Reviewer: How about:

```suggestion
1. Bump the application version on your local machine:
```

  ```bash
  mix nimble_template.bump_version new_version
  ```

> 2. Open a PR to develop.

Reviewer: Minor inconsistency as branches are wrapped into back-ticks below:

```suggestion
2. Open a PR to the `develop` branch.
```

> 3. Once that bump_version PR is merged, create the `release/<version number>` from the `develop` branch, points to the `master` branch.

Reviewer: Minor English corrections:

```suggestion
3. Once the bump_version PR is merged, create the `release/<version number>` from the `develop` branch, points to the `main` branch.
```

Author: as we are now having the Github Action there, will update the Readme content :)
